### PR TITLE
Trigger looker DAG from CI on merge

### PR DESCRIPTION
**Pull request by scholtzan**
Depends on https://github.com/mozilla/telemetry-airflow/pull/2031

This adds a step to the CI to trigger the `looker` DAG when changes get merged into `main`. This way lookml will get deployed automatically

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -126,7 +126,24 @@ jobs:
...
+          workload_identity_pool_id: GCP_WIP_ID
+          workload_identity_pool_provider_id: GCP_WIP_PROVIDER_ID
+          google_project_number: GCP_PROJECT_ID
+          service_account_email: GCP_SERVICE_ACCOUNT_EMAIL
+      - run:
+          name: Trigger looker DAG in Airflow to deploy lookml
+          command: >
+            ID_TOKEN=$(gcloud auth print-identity-token --impersonate-service-account ${GCP_SERVICE_ACCOUNT_EMAIL}) &&
+            DAGRUN_NOTE="DAG triggered by **[${CIRCLE_USERNAME}](https://github.com/${CIRCLE_USERNAME})** from ${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME} CI build [${CIRCLE_BUILD_NUM}](${CIRCLE_BUILD_URL})" &&
+            curl --location --request POST "https://us-west1-moz-fx-telemetry-airflow-prod.cloudfunctions.net/ci-external-trigger"
```
Comment (scholtzan) [inline .circleci/config.yml:143]: todo: this cloud function needs to be updated to take the `dag_id` parameter. Or a separate cloud function needs to be created for this

Comment (scholtzan) [reply .circleci/config.yml:143]: depends on https://github.com/mozilla-it/dataservices-infra/pull/282

File: .circleci/config.yml [hunk 2]
```diff
@@ -126,7 +126,24 @@ jobs:
...
-
+  looker-deploy:
+    executor: gcp-cli/google
+    steps:
+      - gcp-cli/setup:
+          use_oidc: true
+          workload_identity_pool_id: GCP_WIP_ID
+          workload_identity_pool_provider_id: GCP_WIP_PROVIDER_ID
+          google_project_number: GCP_PROJECT_ID
+          service_account_email: GCP_SERVICE_ACCOUNT_EMAIL
```
Comment (mikaeld) [inline .circleci/config.yml:137]:
```suggestion
          workload_identity_pool_id: GCPV2_WORKLOAD_IDENTITY_POOL_ID
          workload_identity_pool_provider_id: GCPV2_CIRCLECI_WORKLOAD_IDENTITY_PROVIDER
          google_project_number: GCPV2_WORKLOAD_IDENTITY_POOL_PROJECT_NUMBER
          service_account_email: GCP_SERVICE_ACCOUNT_EMAIL
```
Requires use of context below (see other comment)

File: .circleci/config.yml [hunk 3]
```diff
@@ -181,3 +198,12 @@ workflows:
           filters:
             branches:
               only: main
+
+      - looker-deploy:
+          name: Trigger looker Airflow DAG to deploy lookml
+          requires:
```
Comment (mikaeld) [inline .circleci/config.yml:204]:
```suggestion
      - looker-deploy:
          name: Trigger looker Airflow DAG to deploy lookml
          context: gcpv2-workload-identity
          requires:
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -181,3 +198,12 @@ workflows:
...
+            - Add commit hash as docker image tag
+          filters:
+            branches:
+              only:
+                - main
```
Comment (mikaeld) [inline .circleci/config.yml:209]:
```suggestion
                - main

```

**Review by mikaeld — APPROVED** [review]
:rocket:

**Pull request merged**